### PR TITLE
boards/mips: remove use of APPDEPS, un-export globally APPDEPS

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR remove the use of APPDEPS to build the pic32 boards. The module generated from the $(CPU_MODEL).S files can be added to the build based on the regulars `USEMODULE` + `DIRS` variables.

There's also no need to export APPDEPS, since it's only used in the global Makefile.include:

https://github.com/RIOT-OS/RIOT/blob/ef4835a0927bff694753435b8fbdc425c0125b6c/Makefile.include#L445

https://github.com/RIOT-OS/RIOT/blob/ef4835a0927bff694753435b8fbdc425c0125b6c/Makefile.include#L549

https://github.com/RIOT-OS/RIOT/blob/ef4835a0927bff694753435b8fbdc425c0125b6c/Makefile.include#L580

As a result, it seems that APPDEPS doesn't need to be exported globally and this PR changes this as well (with a static check)

### Testing procedure

A green Murdock

### Issues/PRs references

None

### CODE REVIEW

File: dist/tools/buildsystem_sanity_check/check.sh [hunk 1]
```diff
@@ -109,9 +109,9 @@ UNEXPORTED_VARIABLES+=('JLINK_DEVICE' 'JLINK_IF')
 UNEXPORTED_VARIABLES+=('JLINK_PRE_FLASH' 'JLINK_RESET_FILE')
 UNEXPORTED_VARIABLES+=('GIT_CACHE' 'GIT_CACHE_DIR')
 UNEXPORTED_VARIABLES+=('LINKXX')
+UNEXPORTED_VARIABLES+=('APPDEPS' 'BUILDDEPS' 'DEBUGDEPS')
```
Comment (fjmolinas) [inline dist/tools/buildsystem_sanity_check/check.sh:112]: 'BUILDDEPS' 'DEBUGDEPS' seem unrelated?

**Comment by fjmolinas** [comment]
I wanted to test this by checking the compiled `hex` but I seem to fail to compile for me, any ideas?

```
"make" -C /data/riotbuild/riotbase/sys/tsrb
"make" -C /data/riotbuild/riotbase/sys/xtimer
objcopy: Unable to recognise the format of the input file `/data/riotbuild/riotbase/examples/asymcute_mqttsn/bin/6lowpan-clicker/asymcute_mqttsn.elf'
objcopy: --change-section-lma .startdata+0xffffffff80000000 never used
objcopy: --change-section-lma .bss+0xffffffff80000000 never used
objcopy: --change-section-lma .data+0xffffffff80000000 never used
objcopy: --change-section-lma .rodata+0xffffffff80000000 never used
objcopy: --change-section-lma .dtors+0xffffffff80000000 never used
objcopy: --change-section-lma .ctors+0xffffffff80000000 never used
objcopy: --change-section-lma .jcr+0xffffffff80000000 never used
objcopy: --change-section-lma .eh_frame+0xffffffff80000000 never used
objcopy: --change-section-lma .fini+0xffffffff80000000 never used
objcopy: --change-section-lma .init+0xffffffff80000000 never used
objcopy: --change-section-lma .text+0xffffffff80000000 never used
objcopy: --change-section-lma .exception_vector+0xffffffff80000000 never used
objcopy: --change-section-lma .bootflash+0xffffffff60000000 never used
/data/riotbuild/riotbase/Makefile.include:591: recipe for target '/data/riotbuild/riotbase/examples/asymcute_mqttsn/bin/6lowpan-clicker/asymcute_mqttsn.bin' failed
make: *** [/data/riotbuild/riotbase/examples/asymcute_mqttsn/bin/6lowpan-clicker/asymcute_mqttsn.bin] Error 1
/home/francisco/workspace/RIOT/makefiles/docker.inc.mk:304: recipe for target '..in-docker-container' failed
make: *** [..in-docker-container] Error 2
make: Leaving directory '/home/francisco/workspace/RIOT/examples/asymcute_mqttsn'

```

**Comment by aabadie** [comment]
> I wanted to test this by checking the compiled hex but I seem to fail to compile for me, any ideas?

I have the same issue but it's the same on master.

File: dist/tools/buildsystem_sanity_check/check.sh (hunk 1 above)
Comment (aabadie) [reply dist/tools/buildsystem_sanity_check/check.sh:112]: Yes but they are also not exported. I added them here as a guard, just in case. I can add them in a separate commit if you prefer?

**Comment by aabadie** [comment]
@fjmolinas, I just noticed that it builds just fine when setting `RIOT_CI_BUILD=1` (with this PR and in master).

- this PR:

```
RIOT_VERSION=test RIOT_CI_BUILD=1 make BOARD=pic32-wifire -C examples/hello-world --no-print-directory 
Building application "hello-world" for "pic32-wifire" with MCU "mips_pic32mz".

   text	   data	    bss	    dec	    hex	filename
 109280	   2780	   7000	 119060	  1d114	/work/riot/RIOT/examples/hello-world/bin/pic32-wifire/hello-world.elf
```

- master:

```
RIOT_VERSION=test RIOT_CI_BUILD=1 make BOARD=pic32-wifire -C examples/hello-world --no-print-directory 
Building application "hello-world" for "pic32-wifire" with MCU "mips_pic32mz".

   text	   data	    bss	    dec	    hex	filename
 109280	   2780	   7000	 119060	  1d114	/work/riot/RIOT/examples/hello-world/bin/pic32-wifire/hello-world.elf
```

**Comment by aabadie** [comment]
Using `git bisect run make BOARD=pic32-wifire -C examples/hello-world` between master and 2020.04, I found that the issue was introduced by 47a5805434c2cc9f1d705bd3327728e2b9cae6ba in #14159.

With the following diff:

```diff
git diff
diff --git a/Makefile.include b/Makefile.include
index 8056a026f3..271b9f847f 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -532,7 +532,7 @@ ifeq ($(RIOT_CI_BUILD),1)
   # Don't build BINFILE on the CI to save some computation time
   BUILD_FILES += $(ELFFILE) $(FLASHFILE)
 else
-  BUILD_FILES += $(ELFFILE) $(BINFILE) $(FLASHFILE)
+  BUILD_FILES += $(ELFFILE) $(FLASHFILE)
 endif
 
 # variables used to compile and link c++
```

The build works fine on master and this PR:

```
RIOT_VERSION=test make BOARD=pic32-wifire -C examples/hello-world/ --no-print-directory 
Building application "hello-world" for "pic32-wifire" with MCU "mips_pic32mz".

"make" -C /work/riot/RIOT/boards/pic32-wifire
"make" -C /work/riot/RIOT/core
"make" -C /work/riot/RIOT/cpu/mips_pic32mz
"make" -C /work/riot/RIOT/cpu/mips32r2_common
"make" -C /work/riot/RIOT/cpu/mips32r2_common/newlib_syscalls_mips_uhi
"make" -C /work/riot/RIOT/cpu/mips32r2_common/periph
"make" -C /work/riot/RIOT/cpu/mips_pic32_common
"make" -C /work/riot/RIOT/cpu/mips_pic32_common/periph
"make" -C /work/riot/RIOT/cpu/mips_pic32mz/p32mz2048efg100
"make" -C /work/riot/RIOT/drivers
"make" -C /work/riot/RIOT/drivers/periph_common
"make" -C /work/riot/RIOT/sys
"make" -C /work/riot/RIOT/sys/auto_init
"make" -C /work/riot/RIOT/sys/stdio_uart
   text	   data	    bss	    dec	    hex	filename
 109280	   2780	   7000	 119060	  1d114	/work/riot/RIOT/examples/hello-world/bin/pic32-wifire/hello-world.elf
```

**Comment by fjmolinas** [comment]
> Using `git bisect run make BOARD=pic32-wifire -C examples/hello-world` between master and 2020.04, I found that the issue was introduced by 47a5805434c2cc9f1d705bd3327728e2b9cae6ba in #14159.

Wow, nice catch, would not have expected that. But why does the `BINFILE` cause the issue?

**Comment by aabadie** [comment]
> But why does the BINFILE cause the issue?

My analysis is that the objcopy used by mips is the default host objcopy, not the toolchain one (because it's broken). And the host objcopy is not able to output the binary:

https://github.com/RIOT-OS/RIOT/blob/e1d27cb7d59af77c1bde54f9090fe39dd6d8425f/cpu/mips_pic32mz/Makefile.include#L14

**Comment by aabadie** [comment]
I think we just skip BINFILE generation for MIPS until we have a working toolchain for it.

**Comment by aabadie** [comment]
see #14405 @fjmolinas

File: dist/tools/buildsystem_sanity_check/check.sh (hunk 1 above)
Comment (fjmolinas) [reply dist/tools/buildsystem_sanity_check/check.sh:112]: Yes, I would prefer a different commit

**Comment by fjmolinas** [comment]
Did as in https://github.com/RIOT-OS/RIOT/pull/14385#issuecomment-651933543 forcing the `HEXFILE` to be built, compared both, no diff.
